File: src/main/java/com/hsmchurch/app/video/ui/response/VideoDetailResponse.java
package com.hsmchurch.app.video.ui.response;

import com.hsmchurch.app.video.domain.BibleContent;
import com.hsmchurch.app.video.domain.VideoType;
import lombok.Builder;
import lombok.Value;

import java.time.LocalDate;
import java.time.LocalDateTime;
import java.util.List;

@Value
@Builder
public class VideoDetailResponse {

    private final Long id;
    private final String youtubeId;
    private final VideoType videoType;
    private final LocalDateTime youtubePublishedAt;
    private final LocalDate filmedAt;
    private final String title;
    private final String preacher;
    private final List<BibleContent> bibleContents;
    private final List<LikeUser> likeUsers;
    private final List<ReplyForVideo> replies;

}

File: src/main/java/com/hsmchurch/app/video/ui/response/VideoDetailResponse.kt
package com.hsmchurch.app.video.ui.response

import com.hsmchurch.app.video.domain.BibleContent
import com.hsmchurch.app.video.domain.VideoType
import lombok.Builder
import lombok.Value

import java.time.LocalDate
import java.time.LocalDateTime

data class VideoDetailResponse(val id: Long,
                               val youtubeId: String,
                               val videoType: VideoType,
                               val youtubePublishedAt: LocalDateTime,
                               val filmedAt: LocalDate,
                               val title: String,
                               val preacher: String?,
                               val bibleContents: List<BibleContent>?,
                               val likeUsers: List<LikeUser>,
                               val replies: List<ReplyForVideo>)
